examples/communication/grandchild_to_grandparent/src/main.rs

```diff
@@ -0,0 +1,145 @@
+use std::rc::Rc;
+use yew::{html, Callback, Component, Context, ContextHandle, ContextProvider, Html};
+
+// This is the shared state between the parent and child components.
+#[derive(Clone, PartialEq)]
+pub struct AppState {
+    value: u32,
+    child_clicked: Callback<()>,
+}
+
+// Our top-level component.
+
+pub struct GrandParent {
+    state: Rc<AppState>,
+}
+
+pub enum Msg {
+    ButtonClick,
+}
+
+impl Component for GrandParent {
+    type Message = Msg;
+    type Properties = ();
+
+    fn create(ctx: &Context<Self>) -> Self {
+        let child_clicked = ctx.link().callback(|_| Msg::ButtonClick);
+        let state = Rc::new(AppState {
+            value: 0,
+            child_clicked,
+        });
+        Self { state }
+    }
+
+    fn update(&mut self, _ctx: &Context<Self>, msg: Self::Message) -> bool {
+        match msg {
+            Msg::ButtonClick => {
+                let value = self.state.value + 1;
+                self.state = Rc::new(AppState {
+                    value,
+                    ..(*self.state).clone()
+                });
```

Comment:
```
Rc::make_mut(&mut self.state).value += 1;
```

Would be more concise to let the clone be implicitly handled by the Rc.

Reply:
Agreed.